### PR TITLE
fix(cmux-sync): stop SecurityAgent prompt storm on go install builds

### LINKED ISSUE #106: cmux-sync enable triggers 10+ SecurityAgent Keychain prompts in rapid succession (go install builds)
**Version:** v0.17.1 (installed via `go install github.com/mvanhorn/agentcookie/cmd/agentcookie@latest`)
**macOS:** 15.x (Darwin 25.5.0)
**Chrome profiles:** 6 (Default + Profile 1–5)

## What happens

Running `agentcookie cmux-sync enable` immediately triggers a loop of SecurityAgent prompts ("agentcookie wants to access the Chrome Safe Storage keychain item") that repeats every ~10 seconds even after clicking Deny. The only escape is killing the launch agent:

```
launchctl unload ~/Library/LaunchAgents/dev.agentcookie.cmux-sync.plist
```

I saw 10+ prompts within the first minute from a single `enable` call.

## Root cause — three compounding bugs

A `go install` build is always ad-hoc signed (no team ID). On such a binary, `SecItemCopyMatching` (the CGO Keychain path) always needs a first-run GUI grant, and three bugs multiply that into a storm:

1. **Goroutine leak in `safeStoragePasswordViaKeybaseFor` (`internal/chrome/keychain_keybase.go`).** `SecItemCopyMatching` is called in a goroutine with a 3s timeout. When the timeout fires the goroutine is abandoned — but the SecurityAgent dialog it triggered stays open. The code then falls through to `security find-generic-password` (the CLI fallback), which opens a *second* dialog. Two simultaneous prompts per Keychain read.

2. **KeepAlive restart loop (`internal/launchd/plist.go`).** The installed plist uses `KeepAlive.SuccessfulExit: false` + `ThrottleInterval: 10`. When the user clicks Deny, `cmux-sync --watch` exits non-zero, launchd waits 10s and restarts it — generating more prompts. Repeats indefinitely.

3. **No Keychain pre-flight in `cmux-sync enable` (`internal/cli/cmux_sync_enable.go`).** `enableCmuxLoop` installs the persistent agent without first verifying Keychain access, so the restart loop starts before the user has any chance to grant access.

## Suggested fixes

- **U1 (CGO skip):** In `SafeStoragePasswordFor`, check `BinaryTeamID(os.Executable())`. If it returns `""` (ad-hoc build), skip `safeStoragePasswordViaKeybaseFor` entirely and go straight to the `security` CLI. `SecItemCopyMatching` on an unsigned binary always requires a GUI prompt; skipping it avoids both the initial prompt and the goroutine leak.
- **U2 (exit 0 on Keychain failure):** In `runCmuxSync`, when `SafeStoragePasswordFor` fails with a Keychain access error in `--watch` mode, print the remediation and exit 0 instead of returning the error, so launchd does not restart.
- **U3 (pre-flight):** In `enableCmuxLoop`, verify Keychain access before installing the agent. If it fails, print `wizard set-keychain-access` instructions and abort without writing the plist.

## Workaround

```
launchctl unload ~/Library/LaunchAgents/dev.agentcookie.cmux-sync.plist
agentcookie wizard set-keychain-access
```

…then re-enable.

---

I have a PR implementing all three fixes ready to open.

### PR DESCRIPTION
**Pull request by henkyermontero**
Closes #106

## What this fixes

Three compounding bugs caused `agentcookie cmux-sync enable` to trigger 10+ SecurityAgent Keychain prompts in rapid succession on `go install` builds (ad-hoc signed, no team ID). See #106 for the full repro and root-cause analysis.

## Changes

- **`internal/chrome/keychain.go` (U1):** `SafeStoragePasswordFor` now checks `BinaryTeamID(os.Executable())` before the CGO path. When it returns `""` (ad-hoc/unsigned binary), it skips `safeStoragePasswordViaKeybaseFor` (SecItemCopyMatching) entirely and goes straight to the `security` CLI. Unsigned binaries always prompt on the CGO path; skipping it avoids both the initial SecurityAgent dialog and the goroutine leaked after the 3s timeout. Signed binaries are unchanged. Adds `IsKeychainAccessError` to classify Keychain-grant failures vs. operational failures. The CGO call is now behind a `safeStorageViaKeybaseRunner` seam for testability.

- **`internal/cli/cmux_sync.go` (U2):** In `--watch` mode, when `SafeStoragePasswordFor` returns a Keychain access error, the agent prints the remediation and exits 0 instead of returning the error — so launchd's `KeepAlive` does not restart it into a prompt loop. Operational errors (cmux down, Chrome DB locked, etc.) still exit non-zero so launchd retries them as before. `--once` mode is unchanged (still returns the error).

- **`internal/cli/cmux_sync_enable.go` (U3):** `enableCmuxLoop` runs a Keychain pre-flight before installing the persistent launch agent and aborts with `wizard set-keychain-access` remediation if it fails — so the restart loop never starts. The pre-flight runs *after* the cmux-not-found no-op, so an absent cmux never triggers a prompt. With U1 in place, on a granted machine the pre-flight and the agent's own read both go through the `security` CLI with no prompt; on an ungranted machine the pre-flight aborts and the agent is never installed.

## Tests

- `internal/chrome/keychain_test.go`: CGO path is skipped for unsigned binaries, used for signed binaries, and a `BinaryTeamID` error is treated as unsigned; `IsKeychainAccessError` boundary cases.
- `internal/cli/cmux_sync_test.go`: `--watch` exits 0 on Keychain access error, returns error on non-Keychain errors, and `--once` returns the error (no exit-0).
- `internal/cli/cmux_sync_enable_test.go`: enable aborts (no mode change, no agent install) when pre-flight fails, proceeds when it passes, and skips the pre-flight entirely when cmux is absent.

`go test ./...` passes.

## Notes / scope

- `BinaryTeamID` shells out to `codesign`; it runs once per process in these call paths, so the added latency is negligible (not cached). Worth a `sync.Once` only if `SafeStoragePasswordFor` ever becomes hot.
- This targets the ad-hoc-signed (`go install`) case, which is the storm in #106. A Developer-ID-signed binary that loses its grant (e.g., after a macOS upgrade) would still take the CGO path; the underlying goroutine-abandonment in `safeStoragePasswordViaKeybaseFor` is out of scope here and could be hardened separately.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/chrome/keychain.go [hunk 1]
```diff
@@ -326,3 +337,20 @@ func IsKeychainLocked(err error) bool {
...
+	if IsKeychainLocked(err) {
+		return true
+	}
+	if err == nil {
+		return false
+	}
+	s := err.Error()
+	return strings.Contains(s, "did you grant access?") ||
+		strings.Contains(s, "not yet in the Safe Storage partition")
+}
```
[the remark this answers is not in the document]
Comment (henkyermontero) [reply internal/chrome/keychain.go:359]: Good catch — fixed in 4cb45dd. `IsKeychainAccessError` no longer matches `-25308`/"interaction is not allowed"; locked-keychain is transient, so `--watch` now exits non-zero on it and lets launchd's KeepAlive retry. Only the genuine missing-grant strings ("did you grant access?", "not yet in the Safe Storage partition") map to exit 0. Test updated to assert the locked cases are NOT access errors.

File: internal/cli/cmux_sync_enable.go [hunk 2]
```diff
@@ -91,6 +99,16 @@ func enableCmuxLoop(cmuxPath string, quiet bool) error {
...
 
+	// Keychain pre-flight: verify Keychain access before installing the agent.
+	// A go install binary is ad-hoc signed and always needs a first-run grant;
+	// failing here prevents the KeepAlive restart loop before it starts.
+	defaultBrowser, _ := chrome.LookupBrowser("")
+	if err := cmuxSyncKeychainCheck(defaultBrowser); err != nil {
+		fmt.Fprintf(os.Stderr, "agentcookie cmux-sync enable: Keychain pre-flight failed — %v\n", err)
+		fmt.Fprintf(os.Stderr, "agentcookie cmux-sync enable: fix first: %s\n", chrome.SafeStorageRemediation)
+		return fmt.Errorf("cmux-sync enable: Keychain not accessible; run `agentcookie wizard set-keychain-access` first")
+	}
```
[the remark this answers is not in the document]
Comment (henkyermontero) [reply internal/cli/cmux_sync_enable.go]: Fixed in 4cb45dd. Moved the pre-flight to after the `cmuxSyncSetMode` `ErrNotFound` guard, immediately before `cmuxSyncInstallAgent`. A never-launched cmux now hits the "launch cmux once" no-op first instead of the Keychain message. Added a test asserting the pre-flight does not run on the `ErrNotFound` path.
